daktari/check.py

```diff
@@ -30,6 +31,11 @@ class Check:
     suggestions: Dict[str, str] = {}
     run_on: Optional[str] = None
 
+    def with_dependencies(self, *dependencies: Type["Check"]) -> "Check":
```

Comment:
Might be good to write a unit test to check `with_dependencies` (maybe a couple of scenarios - one that the dependency gets used, another that the class-level dependencies aren't overwritten?)